Change the detail of reading the Cookie value

TomIsion opened the pull request:
Firstly, I praise your code after I read source code of `cookies.js`. I learned a lot from it and thanks a lot.

Then, I begin use `cookies.js` in my company projects. But I found some questions when I read the cookie like this:

```
MCITY=224-224%3A; 
BD_HOME=1; 
H_PS_PSSID=1468_17948_21113_18559_17001_21454_21409_21377_21525_21190_20928; 
CNUSER=userid=1&&nickName=xxx&loginType=1&token=666;
```

Maybe you will say:

> This `CNUSER` cookie value should be encode.

But this cookie has been used in many other sites in my company, so I could not ask my partner to change this (Maybe the author of this cookie is my boss)

From my partner, they said past days they used [`jquery.cookie.js`](https://github.com/carhartl/jquery-cookie) works well, but nowadays we begin use `React.js`, so I wanna use `Cookies.js` but we met the previous bug. When I read the source code again, I found the question:

Use

```
parts[part[0]] = part.splice(1).join('=');
```

to

```
parts[part[0]] = part[1];
```

I like the use of `Array.prototype.reduce`, so I still wanna keep it.

Also many other ways:

In `jquery.cookie.js`:

```
var cookies = document.cookie ? document.cookie.split('; ') : [],
    i = 0,
    l = cookies.length;
for (; i < l; i++) {
    var parts = cookies[i].split('='),
        name = decode(parts.shift()),
        cookie = parts.join('=');
}
```

Another:

```
var cookies = document.cookie ? document.cookie.split('; ') : [],
    i = 0,
    l = cookies.length;
for (; i < l; i++) {
    var index = cookies[i].indexOf('='),
        name = decode(cookies[i].slice(0, index)),
        cookie = cookies[i].slice(index + 1);
}
```

Hope you can accept my advice or may catch and throw an error when met `=` many times.
Best wishes.

franciscop commented:
Hi @TomIsion , I completely agree that this is a bug but I don't like the way of fixing it so much; I prefer something in the lines of the last example, since it shouldn't be spitted in the same place and this might become confusing in N years when reviewing the code. I'll add a test case and fix it ASAP, thanks for using cookies.js!

franciscop commented:
Nevermind, after reviewing it I find the alternatives to be not so good, so I'll be using your PR, thanks!

franciscop commented:
Version 1.4.2 has just been released in NPM and subsequently on the [CDN](https://unpkg.com/cookiesjs@1.4.2) (I forgot a stray `console.log()` in 1.4.1 so please use the 1.4.2). It includes a test case for this specific situation. If you find any other bug don't hesitate to open an issue.

TomIsion commented:
@franciscop 

Thanks. :tada: 

I found another way to fix this.

Use:

```
.map(function (part) { return part.split('=', 2); })
```

to 

```
.map(function (part) { return part.split('='); })
```

So, you can keep previous code `parts[part[0]] = part[1];`

[String.prototype.split](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split)

Best wishes.
